src/js/view/controls/components/menu/settings-menu.js

```diff
@@ -0,0 +1,345 @@
+    };
+    model.on('change:playbackRates', (changedModel, playbackRates) => {
+        setPlaybackRatesMenu(playbackRates);
+    }, settingsMenu);
+    const setAudioTracksMenu = (audioTracks) => {
+        changeMenuItems(
+            'audioTracks', 
+            audioTracks, 
+            (index) => api.setCurrentAudioTrack(index), 
+            model.get('currentAudioTrack')
+        );
+    };
+    model.on('change:audioTracks', (changedModel, audioTracks) => {
+        setAudioTracksMenu(audioTracks);
+    }, settingsMenu);
+    model.on('change:playbackRate', (changedModel, playbackRate) => {
+        const rates = model.get('playbackRates');
+        let index = -1;
+        if (rates) {
+            index = rates.indexOf(playbackRate);
+        }
+        onMenuItemSelected(settingsMenu.children.playbackRates, index);
+    }, settingsMenu);
+    model.on('change:currentAudioTrack', (changedModel, currentAudioTrack) => {
+        settingsMenu.children.audioTracks.items[currentAudioTrack].activate();
+    }, settingsMenu);
+    model.on('change:playlistItem', () => {
+        // captions.js silently clears captions when the playlist item changes. The reason it silently clear captions
+        // instead of dispatching an event is because we don't want to emit 'captionsList' if the new list is empty.
+        settingsMenu.removeMenu('captions');
+        controlbar.elements.captionsButton.hide();
+
+        // Settings menu should not be visible when switching playlist items via controls or .load()
+        if (settingsMenu.visible) {
+            settingsMenu.close();
+        }
+    }, settingsMenu);
+    model.on('change:playbackRateControls', () => {
+        setPlaybackRatesMenu(model.get('playbackRates'));
+    });
+    model.on('change:castActive', (changedModel, active, previousState) => {
+        if (active === previousState) {
+            return;
+        }
+
+        if (active) {
+            settingsMenu.removeMenu('audioTracks');
+            settingsMenu.removeMenu('quality');
+            settingsMenu.removeMenu('playbackRates');
+        } else {
+            setAudioTracksMenu(model.get('audioTracks'));
+            setLevelsMenu(model.get('levels'));
+            setPlaybackRatesMenu(model.get('playbackRates'));
+        }
+    }, settingsMenu);
+    model.on('change:streamType', () => {
+        setPlaybackRatesMenu(model.get('playbackRates'));
+    }, settingsMenu);
+    window.settingsMenu = settingsMenu;
+    return settingsMenu;
+};
+
+const getColorValue = (option) => colorTable[option];
+const getColorOption = (value) => {
+    const colors = Object.keys(colorTable);
+    let option;
+    for (let i = 0; i < colors.length; i++) {
+        if (colorTable[colors[i]] === value) {
+            option = colors[i];
+            break;
+        }
+    }
+    return option;
+};
+const getPercentOption = (value) => value + '%';
+const getPercentValue = (option) => parseInt(option);
+const captionStyleItems = [
```

Comment:
thoughts on this chunk of captions logic being at the top of the file or maybe a separate file for readability? it's a little tough to parse with it being after all the menu hookup

Reply:
can put it in a utils file